### PR TITLE
Basic Auth Accelerator

### PR DESCRIPTION
**Pull request by ra2085**

### CODE REVIEW

File: basic-auth/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+   * unzip
+   * curl
+   * jq
+   * npm
+
+# (QuickStart) Setup using CloudShell
+
+Use the following GCP CloudShell tutorial, and follow the instructions.
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/ra2085/apigee-usecase-mashup&cloudshell_git_branch=main&cloudshell_workspace=.&cloudshell_tutorial=basic-auth/docs/cloudshell-tutorial.md)
```
Comment (ssvaidyanathan) [inline basic-auth/README.md]: should point to the GoogleCloudPlatform/apigee-samples repo

File: basic-auth/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Setup instructions
+
+1. Clone the apigee-usecase-mashup repo, and switch the basic-auth directory
+
+```bash
+git clone https://github.com/ra2085/apigee-usecase-mashup.git
```
Comment (ssvaidyanathan) [inline basic-auth/README.md]: Why is it pointing to a personal repo?

File: basic-auth/README.md (hunk 1 above)
Comment (ra2085) [reply basic-auth/README.md]: fixed. was testing.

File: basic-auth/README.md (hunk 2 above)
Comment (ra2085) [reply basic-auth/README.md]: fixed. was testing.

File: basic-auth/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Basic Authentication
+
+
+This sample allows you to authenticate an incoming request with a Basic Authentication header using a `USER_ID` and a `USER_PASSWORD` as the encoded credential pair.
```
Comment (ssvaidyanathan) [inline basic-auth/README.md:4]: should we mention its not the most secured way of sending but still a mode of security?

File: basic-auth/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+## Prerequisites
+1. [Provision Apigee X](https://cloud.google.com/apigee/docs/api-platform/get-started/provisioning-intro)
+2. Configure [external access](https://cloud.google.com/apigee/docs/api-platform/get-started/configure-routing#external-access) for API traffic to your Apigee X instance
+3. Access to deploy proxies, create products, apps and developers in Apigee
+4. Make sure the following tools are available in your terminal's $PATH (Cloud Shell has these preconfigured)
+   * [gcloud SDK](https://cloud.google.com/sdk/docs/install)
+   * unzip
+   * curl
+   * jq
+   * npm
```
Comment (ssvaidyanathan) [inline basic-auth/README.md:15]: is npm and/or unzip needed?

File: basic-auth/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+## Setup instructions
+
+1. Clone the apigee-samples repo, and switch the basic-auth directory
+
+```bash
+git clone https://github.com/GoogleCloudPlatform/apigee-samples.git
+cd apigee-samples/basic-auth
+```
+
+2. Edit the `env.sh` and configure the ENV vars. If you don't set variables that are marked as `Optional`, then the deploy script will also provision a mock OIDC authorization server that will allow you to issue JWT access tokens to test this sample
```
Comment (ssvaidyanathan) [inline basic-auth/README.md:32]: need to rewrite this (looks like a paste from the other sample :) )

File: basic-auth/deploy-basic-authn.sh [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+echo " "
+echo "All the Apigee artifacts are successfully deployed!"
+echo " "
+echo "Your USER_ID for testing is: $USER_ID"
+echo " "
+echo "Your USER_PASSWORD for testing is: $USER_PASSWORD"
+echo " "
+echo "Your Basic Authentication header value is: $BASIC_HEADER"
+echo " "
+echo "Your Sample Request URL is: https://$APIGEE_HOST/v1/samples/basic-auth"
```
Comment (ssvaidyanathan) [inline basic-auth/deploy-basic-authn.sh:67]: Is this needed since we have the cURL anyways echoed

File: basic-auth/docs/cloudshell-tutorial.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
+# Basic Authentication
+
+---
+This sample allows you to authenticate an incoming request with a Basic Authentication header using a `USER_ID` and a `USER_PASSWORD` as the encoded credential pair.
```
Comment (ssvaidyanathan) [inline basic-auth/docs/cloudshell-tutorial.md:4]: should we mention its not the most secured way of sending but still a mode of security?

File: basic-auth/README.md (hunk 3 above)
Comment (ra2085) [reply basic-auth/README.md:4]: I'll add a comment in the intro.

File: basic-auth/deploy-basic-authn.sh (hunk 6 above)
Comment (ra2085) [reply basic-auth/deploy-basic-authn.sh:67]: maybe. Just for completion sake. I'm thinking they may want to use a different client.

File: basic-auth/docs/cloudshell-tutorial.md (hunk 7 above)
Comment (ra2085) [reply basic-auth/docs/cloudshell-tutorial.md]: fixed.

File: basic-auth/deploy-basic-authn.sh (hunk 6 above)
Comment (ra2085) [reply basic-auth/deploy-basic-authn.sh:67]: fixed.

File: basic-auth/README.md (hunk 5 above)
Comment (ra2085) [reply basic-auth/README.md]: fixed.

File: basic-auth/README.md (hunk 4 above)
Comment (ra2085) [reply basic-auth/README.md]: fixed.

File: basic-auth/docs/cloudshell-tutorial.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+./deploy-basic-authn.sh
+```
+
+This script creates an API Proxy, an API product, a sample App developer, and an App.
+
+The script output will provide values for `USER_ID`, `USER_PASSWORD` and an encoded Basic Authentication header that you'll be able to use in the next step.
+
+## Testing the sample
+
+ Set the Basic Authentication as a value of the `BASIC_AUTH` environment variable and send a cURL request as shown below (the deployment script already provides a cURL request):
```
Comment (drush1980) [inline basic-auth/docs/cloudshell-tutorial.md:56]: Maybe insert a command to set this header

File: basic-auth/deploy-basic-authn.sh [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+echo " "
+echo "Try this cURL request: curl -H 'Authorization: $BASIC_HEADER' https://$APIGEE_HOST/v1/samples/basic-auth"
```
Comment (drush1980) [inline basic-auth/deploy-basic-authn.sh:69]: add `-v`

**Comment by ssvaidyanathan** [comment]
@ra2085 - whats the update here?
Should we close this PR and re-submit?

**Pull request closed without merging**